OpenRA.Game/Map/CellLayer.cs

```diff
@@ -34,15 +35,35 @@ public CellLayer(TileShape shape, Size size)
 			entries = new T[size.Width * size.Height];
 		}
 
-		public void CopyValuesFrom(CellLayer<T> anotherLayer)
+		// provides a default instance for this type of Layer
+		public static void Init(CellLayer<T> defaultLayer)
 		{
-			if (Size != anotherLayer.Size || Shape != anotherLayer.Shape)
+			defaultCellInfoLayer = defaultLayer;
+		}
+
+        public static void Init(Func<int, int, T> creator, int mapWidth, int mapHeight, TileShape tileShape)
```

Comment:
Please don't do this with a static method! This introduces more global state which we can definitely do without.

You might consider a ctor or instance method that accepted the ```Func<int, int, T> creator``` parameter to set all the cell values if you need a way to initialize the cell values on creation.